### DISCUSSION TITLE
Input in suews_cal_biogenco2()

### DISCUSSION THREAD

**Discussion by XuanhongZhou** [UMEP-dev/UMEP#596] [Q&A]
Dear developers,
  We are using a module of SUEWS: suews_cal_biogenco2(), and we only need to use the functions "co2_biogen()", not the whole SUEWS code. While organizing this function, we had some problems with the input data: we needed to use "gfunc" and "gfunc2" as input data. gfunc is calculated in suews_phys_resist.f95 as "gfunc = gdq*gtemp*gs*gq" (https://suews.readthedocs. io/en/stable/_static/html/suews__phys__resist_8f95_source.html, line 367). Where "gs" again needs to be calculated using the variable "vsmd". In suews_phys_waterdist.f95（https://suews.readthedocs.io/en/stable/_static/html/suews__phys__waterdist_8f95_source.html, line 536）, the calculation process of vsmd requires the participation of another variable, "soilstore_id”, but the variable soilstore_id appears as input data, and we do not find this variable in the input file. So I would like to ask how should I get the value of soilstore_id.
Also, is the value of gfunc2 related to gfunc, we didn't find the calculation process of gfunc2.
Thank you very much for your efforts!

**Comment by biglimp** [comment]
Hopefully @ljarvi or @sunt05 can give you an answer on this.

**Comment by ljarvi** [comment]
Dear @XuanhongZhou. You can set gfunc and gfunc2 to the same gfunc in your case. The gfunc2 refers surface conductance which is calculated from 2 m air temperature and used to calculate the conductance in SUEWS whereas gfunc is calculated from air temperature. In your case, if you use the code separately you have likely only one temperature you want to use the conductance. You are right that soil store is not given as an input but rather calculated in the model from the water balance equation. For surface conductance you need the soil moisture deficit as it is one of the limiting factors for stomata opening. You can naturally here also use observations if available. Hopefully this clarified.

**Reply by XuanhongZhou** [reply]
Thank you very much for your answer, it was very helpful.
I would also like to ask you another question: the input of CO2_biogen has two variables, SnowFrac and LAI_id, which I think correspond to snow and lai in the Meteorological Input File (SSss_YYYY_data_tt.txt). However, the Meteorological Input File gives snow and lai as single values, while SnowFrac and LAI_id are arrays of lengths nsurf and NVegSurf respectively, which I don't quite understand, so I'm looking forward to your answer!

**Reply by XuanhongZhou** [reply]
@ljarvi @sunt05 @biglimp Looking forward to your advice, thanks!

**Reply by sunt05** [reply]
Hi @XuanhongZhou, thanks for your interest in SUEWS. I'm not the lead developer of the CO2 module but I'm here to assist you.

Your questions about `snowfrac` and `lai_id` relate to land cover characterisation in SUEWS, which includes seven types, three of which are vegetated (for more details, please see the SUEWS documentation). To set `snowfrac`, provide an array of seven values; for `lai_id`, supply three. Ensure the ordering matches that specified in the documentation.

Please provide more details about your configuration to better help you.

Good luck with your research!
